File: code/chapter2/benchmark.py
from binary_search import binary_search
from find_pos import find_pos
import timeit
import random

v=list(range(1,10001))

def test_for_loop(n):
  random.seed(2019)
  for _ in range(n):
    find_pos(v,random.randint(1,10000))

def test_bs(n):
  random.seed(2019)
  for _ in range(n):
    binary_search(v,random.randint(1,10000))

# for-loop solution
print(timeit.timeit('test_for_loop(1000)',setup='from __main__ import test_for_loop',number=1))
# binary_search solution
print(timeit.timeit('test_bs(1000)',setup='from __main__ import test_bs',number=1))

File: code/chapter2/benchmark.R
library(microbenchmark)
source('binary_search.R')
source('find_pos.R')

v=1:10000

# call the find_pos 100 times; 
# each time we randomly select an integer as the target value 

# for-loop solution
set.seed(2019)
print(microbenchmark(find_pos(v,sample(10000,1)),times=1000))
# binary-search solution
set.seed(2019)
print(microbenchmark(binary_search(v,sample(10000,1)),times=1000))
